### PR TITLE
Build with scikit build

### PR DESCRIPTION
**Pull request by repagh**
After checking out the updated Slycot, I found that Windows builds were not working for me. Tried hunting down the differences, but could not get sufficient control over the Fortran side of the build process with numpy/setuptools. So I ended up implementing a pretty large change in the build system, using scikit-build to replace the numpy/setuptools based build system. 

scikit-build wraps setuptools, and combines with CMake for the Fortran and C side of the builds. This enables more fine-grained control over the Fortran compilation, and selection of appropriate build generators. For Windows this uses NMake and Flang, for Linux this uses gnu make and fortran, for Mac it combines gfortran with clang and again gnu make. 

Note: somehow the commit messages got mixed up? I did a merge+squash on a new branch, had proper messages locally, but here I have a bunch of old messages???

commit be8cdfa9ba3ccd4c131e79d8e990a96e5deae4ca (HEAD -> build-with-scikit-build, origin/build-with-scikit-build)
Author: Rene van Paassen <Rene.vanPaassen@gmail.com>
Date:   Wed Jul 11 22:24:49 2018 +0200

    Add a comment on the many manual selections needed for Windows build.

commit 2ee5339d3fc4a4016c3102e0f8b143cbf2599b75
Author: Rene van Paassen <Rene.vanPaassen@gmail.com>
Date:   Wed Jul 11 22:14:55 2018 +0200

    Conversion to building with scikit-build. Uses cmake for better
    autodetection of available libraries (openblas/lapack)
    
    - deleted Slycot/slycot/setup.py
    - CMakeLists.txt in Slycot and Slycot/slycot
    - adapted Slycot/setup.py
    
    * Windows build needs specification of generator, e.g.
      python setup.py build -G "NMake Makefiles"
    * OSX and Linux builds select proper generator automatically

### CODE REVIEW

File: conda-recipe-openblas/meta.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ requirements:
     - numpy
     - openblas >=0.3.0
     - libflang  # [win]
-    - libgfortran # [not win]
+    - libgfortran # [not win]
```
Comment (murrayrm) [inline conda-recipe-openblas/meta.yaml]: Why extra spaces?

File: conda-recipe/meta.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ requirements:
     - numpy
     - lapack
     - libflang  # [win]
-    - libgfortran # [not win]
+    - libgfortran # [not win]
```
Comment (murrayrm) [inline conda-recipe/meta.yaml]: Why extra spaces?

**Comment by murrayrm** [comment]
Travis CI failed.  Looks like it can't find `skbuild`.

Will try this out on my local machine when I get a chance (probably weekend, soonest).

File: conda-recipe-openblas/meta.yaml (hunk 1 above)
Comment (repagh) [reply conda-recipe-openblas/meta.yaml]: dunno. Editing on windows gives me that stuff sometimes.... Will remove them.

**Comment by repagh** [comment]
I forgot that I used pip to install a recent scikit-build. 

I see people are now working on a new release of scikit-build for conda-forge. I think it is best to pick things up when that happened.

**Comment by moorepants** [comment]
It may work if you just add `pip install scikit-build` here: https://github.com/python-control/Slycot/blob/master/.travis.yml#L67

**Comment by repagh** [comment]
Closing, superseded by PR #38

**Pull request closed without merging**